tests/end_to_end/test_kakarot.py

```diff
@@ -0,0 +1,148 @@
+                int(x)
+                for x in params["stack"]
+                .format(
+                    account_address=addresses[0].address,
+                    timestamp=result.block_timestamp,
+                    block_number=result.block_number,
+                )
+                .split(",")
+            ]
+            if params["stack"]
+            else []
+        )
+        assert memory_result == hex_string_to_bytes_array(params["memory"])
+
+        events = params.get("events")
+        if events:
+            # Events only show up in a transaction, thus we run the same call, but in a tx
+            tx = await call.invoke(max_fee=max_fee)
+            await tx.wait_for_acceptance()
+            receipt = await starknet.get_transaction_receipt(tx.hash)
+            assert receipt.status == TransactionStatus.ACCEPTED_ON_L2
+            assert [
+                [
+                    # we remove the key that is used to convey the emitting kakarot evm contract
+                    event.keys[1:],
+                    event.data,
+                ]
+                for event in receipt.events
+                if event.from_address != eth.address
+            ] == events
+
+    class TestComputeStarknetAddress:
+        async def test_should_return_same_as_deployed_address(
+            self, compute_starknet_address, addresses
+        ):
+            eoa = addresses[0]
+            starknet_address = await compute_starknet_address(eoa.address)
+            assert eoa.starknet_contract.address == starknet_address
+
+    class TestDeployExternallyOwnedAccount:
+        async def test_should_deploy_starknet_contract_at_corresponding_address(
+            self,
+            fund_starknet_address,
+            deploy_externally_owned_account,
+            compute_starknet_address,
+            get_contract,
+        ):
+            evm_address = generate_random_evm_address()
+            starknet_address = await compute_starknet_address(evm_address)
+            await fund_starknet_address(starknet_address, PRE_FUND_AMOUNT / 1e18)
+
+            await deploy_externally_owned_account(evm_address)
+            eoa = await get_contract(
+                "externally_owned_account", address=starknet_address
+            )
+            actual_evm_address = (
+                await eoa.functions["get_evm_address"].call()
+            ).evm_address
+            assert actual_evm_address == int(evm_address, 16)
+
+        async def test_should_send_fees_to_caller(
+            self,
+            starknet: FullNodeClient,
+            fund_starknet_address,
+            deploy_externally_owned_account,
+            compute_starknet_address,
+            deployer: Account,
+            ethBalanceOf,
+            deploy_fee: int,
+        ):
+            # using a different seed here so that the evm address is different from the one used in the previous test
+            evm_address = generate_random_evm_address(seed=0xDEADBEEF)
```

Comment:
Would it be useful to have a counter for this seed? This would avoid people adding tests to mistakenly reusing the same seed.

Reply:
we could definitely use a counter sure, it was `seed=1` in the original file and I put this more obvious value instead. I created an issue https://github.com/kkrt-labs/kakarot/issues/721